### PR TITLE
Improve attestation capability and interface

### PR DESCRIPTION
**Pull request by jxyang**
including:

1. Define an environment variable `MYST_WANT_TLS_CREDENTIAL` to allow users to specify their different requirements on TLS credentials. 
1. Move syscallext.h to tee.h so users can make extended syscalls with no fixed syscall numbers.
1. Add certificate-related functions
1. Define a new syscall that generates cert, private key, and public-key-attesting report and dump them on tmpfs.
1. Update the syscall SYS_myst_free_creds to it additionally free the report buffer.
1. Clean up the previous implementation of gencreds.

### CODE REVIEW

**Review by mikbras — COMMENTED** [review]
I took a first pass. Overall it looks good. I wonder whether renaming <syscallext.h> to <tee.h> will be confusing or not. These extended syscalls apply to the non-tee case is some instance (Linux). 

Anyway, I will take a closer look tomorrow.

**Comment by jxyang** [comment]
I agree "tee.h" is not the best name. I keep the name to be compatible with existing customer usages.

File: doc/user-getting-started-tee-aware.md [hunk 1]
```diff
@@ -203,3 +205,107 @@ application running inside a TEE with no regard to the application identity.
 If the application does want to reject/approve based on app identity, it
 must include `myst/tee.h`, define a function similar to `_verifier`,
 and pass it to syscall `SYS_myst_verify_cert`.
+
+## Write a C# program that obtains attestation credentials statically
+
+High level languages like C# don't have direct access to syscalls as
+C/C++ do. Therefore, we introduce the environment variable
```
Comment (mikbras) [inline doc/user-getting-started-tee-aware.md:212]: do => does

File: include/myst/tee.h [hunk 2]
```diff
@@ -4,6 +4,52 @@
 #ifndef _MYST_TEE_H
 #define _MYST_TEE_H
 
+#include <stdint.h>
+
+/* Extended syscalls of Mystikos */
+enum
+{
```
Comment (mikbras) [inline include/myst/tee.h:11]: Not all of these syscalls should be exposed. For example
```
    SYS_myst_trace = 1001,
    SYS_myst_trace_ptr = 1002,
    SYS_myst_dump_stack = 1003,
    SYS_myst_dump_ehdr = 1004,
    SYS_myst_dump_argv = 1005,
    SYS_myst_add_symbol_file = 1006,
    SYS_myst_load_symbols = 1007,
    SYS_myst_unload_symbols = 1008,
    SYS_myst_clone = 1012,
    SYS_myst_gcov_init = 1013,
    SYS_myst_max_threads = 1014,
    SYS_myst_poll_wake = 1015,
    SYS_myst_run_itimer = 1016,
```

**Review by mikbras — APPROVED** [review]

**Pull request merged**